dmriprep/interfaces/images.py

```diff
@@ -0,0 +1,131 @@
+"""Image tools interfaces."""
+
+from nipype import logging
+from nipype.interfaces.base import (
+    traits, TraitedSpec, BaseInterfaceInputSpec, SimpleInterface, File
+)
+
+LOGGER = logging.getLogger('nipype.interface')
+
+
+class ExtractB0InputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc='dwi file')
+    b0_ixs = traits.List(traits.Int, mandatory=True,
+                         desc='Index of b0s')
+
+
+class ExtractB0OutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc='b0 file')
+
+
+class ExtractB0(SimpleInterface):
+    """
+    Extract all b=0 volumes from a dwi series.
+
+    Example
+    -------
+
+    >>> os.chdir(tmpdir)
+    >>> extract_b0 = ExtractB0()
+    >>> extract_b0.inputs.in_file = str(data_dir / 'dwi.nii.gz')
+    >>> extract_b0.inputs.b0_ixs = [0, 1, 2]
+    >>> res = extract_b0.run()  # doctest: +SKIP
+
+    """
+    input_spec = ExtractB0InputSpec
+    output_spec = ExtractB0OutputSpec
+
+    def _run_interface(self, runtime):
+        self._results['out_file'] = extract_b0(
+            self.inputs.in_file,
+            self.inputs.b0_ixs,
+            newpath=runtime.cwd)
+        return runtime
+
+
+def extract_b0(in_file, b0_ixs, newpath=None):
+    """Extract the *b0* volumes from a DWI dataset."""
+    import numpy as np
+    import nibabel as nib
+    from nipype.utils.filemanip import fname_presuffix
+
+    out_file = fname_presuffix(
+        in_file, suffix='_b0', newpath=newpath)
+
+    img = nib.load(in_file)
+    data = img.get_fdata()
```

Comment:
@arokem can we avoid the internal cast to float64 here? I believe float32 should be enough for the average b0, right?
```suggestion
    data = img.get_fdata(dtype='float32')
```